{"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.10.14","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"none","dataSources":[{"sourceId":2232808,"sourceType":"datasetVersion","datasetId":721}],"dockerImageVersionId":30804,"isInternetEnabled":true,"language":"python","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# Introduction\r\n## About Authur\r\n* I am Masod ul Hassan Ghauri. I am a student of Data science & AI subject.\r\n* LnkDn Profile: https://www.linkedin.com/in/masod-ghauri-914a18334/\r\n* Kaggle Profile: https://www.kaggle.com/masodulhassanghauri\r\n* GitHub Profile: https://github.com/Masod6\n# About Data set\nThis data set download from kaggle https://www.kaggle.com/datasets/zusmani/mygenome?select=Child+1+Genome.csv\nZeeshan-ul-hassan Usmani’s Genome Phenotype SNPs Raw Data\n\nGenomics is a branch of molecular biology that involves structure, function, variation, evolution and mapping of genomes. There are several companies offering next generation sequencing of human genomes from complete 3 billion base-pairs to a few thousand Phenotype SNPs. I’ve used 23andMe (using Illumina HumanOmniExpress-24) for my DNA’s Phenotype SNPs.","metadata":{}},{"cell_type":"code","source":"# import libraries\nimport numpy as np\nimport pandas as pd\nimport  matplotlib.pyplot as plt\nimport seaborn as sns\nimport warnings\n# Ignore all warnings\nwarnings.filterwarnings(\"ignore\")","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-14T18:00:01.472967Z","iopub.execute_input":"2024-12-14T18:00:01.473413Z","iopub.status.idle":"2024-12-14T18:00:01.479090Z","shell.execute_reply.started":"2024-12-14T18:00:01.473360Z","shell.execute_reply":"2024-12-14T18:00:01.477851Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df=pd.read_csv('/kaggle/input/mygenome/Child 1 Genome.csv')","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-14T18:08:09.775086Z","iopub.execute_input":"2024-12-14T18:08:09.775457Z","iopub.status.idle":"2024-12-14T18:08:10.396899Z","shell.execute_reply.started":"2024-12-14T18:08:09.775423Z","shell.execute_reply":"2024-12-14T18:08:10.395880Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.head(10)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-14T18:08:26.965308Z","iopub.execute_input":"2024-12-14T18:08:26.965716Z","iopub.status.idle":"2024-12-14T18:08:26.988669Z","shell.execute_reply.started":"2024-12-14T18:08:26.965666Z","shell.execute_reply":"2024-12-14T18:08:26.987265Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.info()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.describe()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.isnull().sum()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.duplicated()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.head()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Visualize the number of SNPs per chromosome\nChromosomes with higher/lower SNP counts indicate regions of dense genetic information or gaps.","metadata":{}},{"cell_type":"code","source":"# SNPs distribution\ndf['chromosome'].value_counts().plot(kind='bar')\nplt.show()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Identify the frequency of each allele (e.g., A, G, T, C) in the dataset\nThis chart shows which alleles are most common in the dataset\n","metadata":{}},{"cell_type":"code","source":"from collections import Counter\nimport matplotlib.pyplot as plt\n\n# Count alleles\nalleles = ''.join(df['genotype'].dropna())\nallele_count = Counter(alleles)\n\n# Plot\nplt.bar(allele_count.keys(), allele_count.values(), color='skyblue')\nplt.xlabel('Alleles')\nplt.ylabel('Frequency')\nplt.title('Allele Frequency Distribution')\nplt.show()\n\n\n","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Visualize the number of SNPs per chromosome\nChromosomes with higher/lower SNP counts indicate regions of dense genetic information or gaps.\n","metadata":{}},{"cell_type":"code","source":"\nchrom_counts = df['chromosome'].value_counts()\n\n# Bar plot for chromosome distribution\nchrom_counts.plot(kind='bar', color='purple', alpha=0.7)\nplt.xlabel('Chromosome')\nplt.ylabel('Number of SNPs')\nplt.title('SNP Distribution Across Chromosomes')\nplt.show()\n","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Show proportion of SNPs associated with disease risk.\nThis chart highlights what percentage of the genome might contribute to disease susceptibility.\n","metadata":{}},{"cell_type":"code","source":"\n# Filter disease-related SNPs\nrisk_snps = df[df['# rsid'] == df['genotype']]\n\n# Create pie chart\nlabels = ['Risk SNPs', 'Non-Risk SNPs']\nsizes = [len(risk_snps), len(df) - len(risk_snps)]\ncolors = ['red', 'green']\n\nplt.pie(sizes, labels=labels, colors=colors, autopct='%1.1f%%', startangle=90)\nplt.title('Proportion of Disease-Associated SNPs')\nplt.show()\n# This chart highlights what percentage of the genome might contribute to disease susceptibility.","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Visualize the frequency of each genotype (e.g., AA, AG, GG).\nThis can indicate whether homozygous or heterozygous genotypes are more prevalent.\n","metadata":{}},{"cell_type":"code","source":"\ngenotype_counts = df['genotype'].value_counts()\n\n# Bar plot for genotype counts\ngenotype_counts.plot(kind='bar', color='orange', alpha=0.8)\nplt.xlabel('Genotype')\nplt.ylabel('Frequency')\nplt.title('Genotype Frequency Distribution')\nplt.show()\n","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"plt.figure(figsize=(10, 6))\nsns.scatterplot(data=df, x='position', y='chromosome', hue='# rsid', palette='cool', alpha=0.6)\nplt.title('Chromosomal SNP Positions')\nplt.xlabel('Position on Chromosome')\nplt.ylabel('Chromosome')\nplt.legend(title='Risk Allele')\nplt.show()","metadata":{},"outputs":[],"execution_count":null},{"cell_type":"code","source":"df.head(1)","metadata":{},"outputs":[],"execution_count":null}]}